File: RepoTransBench/target_projects/Java/Go/alexjlockwood_adp-path-morph-play-to-pause/public_tests/play_to_pause_morph_additional_public_test.go
package public_tests

import (
	"testing"

	"pathmorph"
)

func TestMultipleTogglesPublic(t *testing.T) {
	morph := pathmorph.NewPlayToPauseMorph(true)
	expected := true
	for i := 0; i < 4; i++ {
		morph.Toggle()
		expected = !expected
		if morph.IsPlaying() != expected {
			t.Errorf("Toggle #%d: want playing=%v, got %v", i+1, expected, morph.IsPlaying())
		}
	}
}

func TestGetStateStringsPublic(t *testing.T) {
	morph := pathmorph.NewPlayToPauseMorph(false)
	if got := morph.GetState(); got != "PAUSE" {
		t.Errorf("Initial GetState: want PAUSE, got %s", got)
	}
	morph.Toggle()
	if got := morph.GetState(); got != "PLAY" {
		t.Errorf("After toggle GetState: want PLAY, got %s", got)
	}
}

func TestEdgeCaseNoTogglePublic(t *testing.T) {
	morphPause := pathmorph.NewPlayToPauseMorph(false)
	if got := morphPause.GetState(); got != "PAUSE" {
		t.Errorf("morphPause: want PAUSE, got %s", got)
	}
	morphPlay := pathmorph.NewPlayToPauseMorph(true)
	if got := morphPlay.GetState(); got != "PLAY" {
		t.Errorf("morphPlay: want PLAY, got %s", got)
	}
}

File: RepoTransBench/target_projects/Java/C#/alexjlockwood_adp-path-morph-play-to-pause/public_tests/PlayToPauseMorphAdditionalPublicTest.cs
using Xunit;
using Adp.PathMorph;

namespace Adp.PathMorph.PublicTests
{
    public class PlayToPauseMorphAdditionalPublicTest
    {
        [Fact]
        public void TestMultipleTogglesPublic()
        {
            var morph = new PlayToPauseMorph(true);
            // Toggle 4 times, state should alternate each time
            bool expected = true;
            for (int i = 0; i < 4; i++)
            {
                morph.Toggle();
                expected = !expected;
                Assert.Equal(expected, morph.IsPlaying());
            }
        }

        [Fact]
        public void TestGetStateStringsPublic()
        {
            var morph = new PlayToPauseMorph(false);
            Assert.Equal("PAUSE", morph.GetState());
            morph.Toggle();
            Assert.Equal("PLAY", morph.GetState());
        }

        [Fact]
        public void TestEdgeCaseNoTogglePublic()
        {
            var morphPause = new PlayToPauseMorph(false);
            Assert.Equal("PAUSE", morphPause.GetState());
            var morphPlay = new PlayToPauseMorph(true);
            Assert.Equal("PLAY", morphPlay.GetState());
        }
    }
}